designs/Commands/clear-package.md

```diff
@@ -0,0 +1,92 @@
+# Clearing an specific package from NuGet cache
+
+* Status: **Incubating**
+* Author(s): [Fernando Aguilar](https://github.com/dominofire)
+
+The purpose is to add a feature to `nuget locals` command to delete a package from NuGet cache folder
+
+
+## Problem Background
+
+There's some interest on having features to delete an specific package instead of clearing the whole cache.
+
+- https://github.com/NuGet/Home/issues/5713 : Comments on deleting specific packages
+- https://github.com/NuGet/Home/issues/4980 : Comments on cache handling polices
+- https://github.com/chgill-MSFT/NuGetCleaner : A clean-up tool relying on last-acesss file metadata in NTFS file system
+
+In PackageReference world, the global packages folders are shared across all restored projects. 
+
+There's no current API or command to delete an specific package.
+
+
+## Who are the customers
+
+All `dotnet nuget locals` users.
+
+
+## Goals
+
+- Add command options to `dotnet nuget locals` to delete _one_ package from a NuGet cache folder, either _one_ or multiple versions.
+
+
+## Non-Goals
+
+- Cache expiration policies are not covered
+- Deleting cached packages from an specific folder are not covered 
+- `packages.config` projects are not covered. To delete a package, delete the pacakge folder from `packages` folder
+- `nuget.exe` commands are not covered
+
+
+## Solution Overview
+
+From ideas described in [this comment](https://github.com/NuGet/Home/issues/5713#issuecomment-320560636), the following command options will delete an specific package. Exmples are listed below.
+
+Delete a package from cache with an specific version:
+
+```
+dotnet nuget locals global-packages --clear --package a --version 1.3
+```
+
+Delete all package versions from cache:
+
+```
+dotnet nuget locals global-packages --clear --package a
+```
+
+Delete all versions of multiple packages:
+
+```
+dotnet nuget locals http-cache --clear --package a
+dotnet nuget locals http-cache --clear --package b
+dotnet nuget locals http-cache --clear --package c
+dotnet nuget locals http-cache --clear --package d
+```
+
+Delete all versions of multiple packages (version 2):
+
+```
+dotnet nuget locals global-packages -clear -packages a,b,c,d,e
```

Comment:
Does this mean that `--version` is only valid with `--package` and not `--packages`? Should this be mentioned explicitly?

If it is valid, what's the expected behaviour? Is it possible to specify both multiple packages and multiple versions?

If we don't have customers asking to be able to delete multiple packages in a single command, I suggest it's better to drop this feature, and wait for customer feedback, with information to help us understand why it's difficult to delete one package at a time, like your example above. We can add features/arguments, but removing them is a breaking change, so much more difficult.

Reply:
you can use the same double colon syntax for version{s} as dotnet-new template. https://www.nuget.org/packages/NUnit3.DotNetNew.Template/ `dotnet new --install NUnit3.DotNetNew.Template::1.9.0`. it is ugly, it doesn't "fit" with anything else in the entire ecosystem... but (that didn't stopped it becoming part of dotnet-new so) at least you have a precedence there. :)